quickannotator/api/v1/utils/shared_crud.py

```diff
@@ -131,32 +226,44 @@ def __init__(self, image_id: int, annotation_class_id: int, is_gt: bool, in_work
             self.model = create_dynamic_model(build_annotation_table_name(image_id, annotation_class_id, is_gt=is_gt))
 
     # CREATE
-    # TODO: make this function applicable to predicted annotations, not just ground truths.
-    def insert_annotations(self, polygons: List[BaseGeometry]) -> List[models.Annotation]:
+    # TODO: consider adding optional parameter to allow tileids to be passed in.
+    def insert_annotations(self, polygons: List[BaseGeometry], tile_ids: List[int] | int=None) -> List[models.Annotation]:
+        # Initial validation
+        if len(polygons) == 0:
+            return []
+        
+        if isinstance(tile_ids, int):
+            tile_ids = [tile_ids] * len(polygons)
+        
+        if tile_ids is not None and len(polygons) != len(tile_ids):
+            raise ValueError("The lengths of polygons and tile_ids must match if tile_ids is a list.")
+
         # in_work_mag is true because we expect the polygons are scaled at this point.
         tilespace = get_tilespace(self.image_id, self.annotation_class_id, in_work_mag=True)
         new_annotations = []
-        for polygon in polygons:
+
+        for i, polygon in enumerate(polygons):
             scaled_polygon = self.scale_polygon(polygon, self.scaling_factor)
-            tile_id = tilespace.point_to_tileid(scaled_polygon.centroid.x, scaled_polygon.centroid.y)
+
+            # Determine tile_id based on provided parameters or calculate it
+            if tile_ids is not None:
+                current_tile_id = tile_ids[i]
```

Comment:
this freaks me out a bit - this is not pythonic - a normal way would be to use a 
```python
for current_tile_id, polygon in zip(polygon,tile_ids):
```

which removes the chances for *a lot* of bugs. is that pattern applicable here?

Reply:
Done. Looks much better now!